src/fiatconnect/TransferStatusScreen.test.tsx

```diff
@@ -0,0 +1,105 @@
+import { createMockStore, getMockStackScreenProps } from 'test/utils'
+import { mockFiatConnectTransfers, mockFiatConnectQuotes } from 'test/values'
+import { Screens } from 'src/navigator/Screens'
+import { fireEvent, render } from '@testing-library/react-native'
+import { Provider } from 'react-redux'
+import FiatConnectQuote from 'src/fiatExchanges/quotes/FiatConnectQuote'
+import { FiatAccountType } from '@fiatconnect/fiatconnect-types'
+import { CICOFlow } from 'src/fiatExchanges/utils'
+import { FiatConnectQuoteSuccess } from 'src/fiatconnect'
+import TransferStatusScreen from 'src/fiatconnect/TransferStatusScreen'
+import { navigate, navigateBack, navigateHome } from 'src/navigator/NavigationService'
+import React from 'react'
+import networkConfig from 'src/web3/networkConfig'
+import i18n from 'i18next'
+
+describe('TransferStatusScreen', () => {
+  const mockStore = (overrides: any = {}) => {
+    const store = createMockStore({
+      fiatConnect: {
+        transfer: mockFiatConnectTransfers[0],
+        ...overrides,
+      },
+    })
+    store.dispatch = jest.fn()
+    return store
+  }
+
+  const mockQuote = new FiatConnectQuote({
+    flow: CICOFlow.CashOut,
+    fiatAccountType: FiatAccountType.BankAccount,
+    quote: mockFiatConnectQuotes[1] as FiatConnectQuoteSuccess,
+  })
+
+  const mockScreenProps = () =>
+    getMockStackScreenProps(Screens.FiatConnectTransferStatus, {
+      flow: CICOFlow.CashOut,
+      fiatAccount: {
+        fiatAccountId: 'some-fiat-account-id',
+        accountName: 'some-friendly-name',
+        institutionName: 'some-bank',
+        fiatAccountType: FiatAccountType.BankAccount,
+      },
+      normalizedQuote: mockQuote,
+    })
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe('success view', () => {
+    it('navigates home on success', () => {
+      const store = mockStore()
+      const { queryByTestId, getByTestId } = render(
+        <Provider store={store}>
+          <TransferStatusScreen {...mockScreenProps()} />
+        </Provider>
+      )
+      expect(queryByTestId('Continue')).toBeTruthy()
+      fireEvent.press(getByTestId('Continue'))
+      expect(navigateHome).toHaveBeenCalledWith()
+    })
+
+    it('shows TX details on Celo Explorer on success', () => {
+      const store = mockStore()
+      const { queryByTestId, getByTestId } = render(
+        <Provider store={store}>
+          <TransferStatusScreen {...mockScreenProps()} />
+        </Provider>
+      )
+      expect(queryByTestId('txDetails')).toBeTruthy()
+      fireEvent.press(getByTestId('txDetails'))
+      expect(navigate).toHaveBeenCalledWith(Screens.WebViewScreen, {
+        uri: `${networkConfig.celoExplorerBaseTxUrl}${mockFiatConnectTransfers[0].txHash}`,
+      })
+    })
+  })
+
+  describe('failure view', () => {
```

Comment:
I had some trouble testing out the behavior of the header buttons here. I'm dynamically adjusting the header components on the fly depending on the result of the transfer selector, but when selecting any header components by test ID, they're not found in the tests (despite working as expected in-app). Would love some advice on this if anyone has any insight!

Reply:
this is likely because navigation prop is mocked in `getMockScreenProps`, so headers are not rendered. Could you just check navigation.setOptions mock to check if it's called with the correct params?